### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Tarasulko**
[DEMO LINK](https://tarasulko.github.io/react_counter/)

### CODE REVIEW

**Review by Artem5457 — CHANGES_REQUESTED** [review]

File: src/components/Counter.scss [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    border-color: #711eca;
+    border-radius: 10px;
+
+    transition: background-color 0.3s;
+  }
+
+  &__button:hover {
+    background-color: #fff;
+  }
+}
```
Comment (aliceblunt8) [inline src/components/Counter.scss:21]: add empty file at the end of the file

File: src/components/Counter.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+import './Counter.scss';
+
+type State = {
+  number: number,
+};
+
+class Counter extends React.Component<{}, State> {
+  state = {
```
Comment (aliceblunt8) [inline src/components/Counter.tsx:10]:
```suggestion
  state: State = {
```

File: src/components/Counter.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    number: 0,
+  };
+
+  addOne = () => {
+    this.setState((currState) => ({ number: currState.number + 1 }));
```
Comment (aliceblunt8) [inline src/components/Counter.tsx:15]:
```suggestion
    this.setState((prevState) => ({ number: prevState.number + 1 }));
```

File: src/components/Counter.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    this.setState((currState) => ({ number: currState.number + 100 }));
+  };
+
+  addOneOr100 = () => {
+    const remainder = this.state.number % 5;
+
+    if (remainder === 0) {
+      this.add100();
+
+      return;
```
Comment (aliceblunt8) [inline src/components/Counter.tsx:28]: remove return, so it always add one initially or after adding 100

```suggestion
```

File: src/components/Counter.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  render() {
+    return (
+      <div className="Counter">
+        <h1 className="Counter__title">
+          Current number:
+          {this.state.number}
+        </h1>
+        <button type="button" className="Counter__button" onClick={this.addOne}>
+          Add 1
+        </button>
```
Comment (aliceblunt8) [inline src/components/Counter.tsx:43]: move props to the new line when they're more than 2 of them

```suggestion
        <button
          type="button"
          className="Counter__button"
          onClick={this.addOne}
        >
          Add 1
        </button>
```

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/components/Counter.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        <button type="button"
+          className="Counter__button"
+          onClick={this.add100}
+        >
+          Add 100
+        </button>
+        <button type="button"
+          className="Counter__button"
+          onClick={this.addOneOr100}
+        >
```
Comment (aliceblunt8) [inline src/components/Counter.tsx:55]:
```suggestion
        <button
          type="button"
          className="Counter__button"
          onClick={this.add100}
        >
          Add 100
        </button>
        <button
          type="button"
          className="Counter__button"
          onClick={this.addOneOr100}
        >
```

**Review by aliceblunt8 — APPROVED** [review]

**Pull request closed without merging**